File: For pub and sub/siena-0.4.3/siena-0.4.3/tests/TestTokenizer.cc
// -*- C++ -*-
//
//  This file is part of Siena, a wide-area event notification system.
//  See http://www.cs.colorado.edu/serl/dot/siena.html
//
//  Author: Antonio Carzaniga <carzanig@cs.colorado.edu>
//  See the file AUTHORS for full details. 
//
//  Copyright (C) 1998-2000 University of Colorado
//
//  This program is free software; you can redistribute it and/or
//  modify it under the terms of the GNU General Public License
//  as published by the Free Software Foundation; either version 2
//  of the License, or (at your option) any later version.
//
//  This program is distributed in the hope that it will be useful,
//  but WITHOUT ANY WARRANTY; without even the implied warranty of
//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
//  GNU General Public License for more details.
//
//  You should have received a copy of the GNU General Public License
//  along with this program; if not, write to the Free Software
//  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307,
//  USA, or send email to serl@cs.colorado.edu.
//
//
// $Id: TestTokenizer.cc,v 1.2 2003/07/09 20:43:17 carzanig Exp $
//
#include <iostream>
#include <exception>

#include <siena/SENP.h>

//
// this gives you access to the Siena headers (and to the Siena namespace)
//
#include "SENPTokenizer.h"

int main (int argc, char *argv[])
{
    string line;
    char buf[1024];

    try {
	while(cin.getline(buf, 1024)) {
	    buf[1023]='\0';
	    string line(buf);
	    SENPTokenizer t(line);
	    int tok;
	    while((tok = t.nextToken()) != SENPTokenizer::T_EOF) {
		switch(tok) {
		case SENPTokenizer::T_INT:
		    cout << "int: " << t.ival << endl;
		    break;
		case SENPTokenizer::T_DOUBLE:
		    cout << "double: " << t.dval << endl;
		    break;
		case SENPTokenizer::T_STR:
		    cout << "str: " << t.sval << endl;
		    break;
		case SENPTokenizer::T_ID:
		    cout << "id: " << t.sval << endl;
		    break;
		case SENPTokenizer::T_BOOL:
		    cout << "bool: " << t.bval << endl;
		    break;
		case SENPTokenizer::T_OP:
		    cout << "op: " + SENP::Operators[t.oval] << endl;
		    break;
		case SENPTokenizer::T_LPAREN:
		    cout << "{" << endl;
		    break;
		case SENPTokenizer::T_RPAREN:
		    cout << "}" << endl;
		    break;
		case SENPTokenizer::T_UNKNOWN:
		    cout << "unknown" << endl;
		    break;
		}
	    }
	}
    } catch (exception &ex) {
	cerr << "error: " << ex.what() << endl;
    }
}

File: Sequential/siena-2.0.4/test/sienatest/TestTokenizer.java
//
//  This file is part of Siena, a wide-area event notification system.
//  See http://www.inf.usi.ch/carzaniga/siena/
//
//  Author: Antonio Carzaniga (firstname.lastname@usi.ch)
//  See the file AUTHORS for full details. 
//
//  Copyright (C) 1998-2002 University of Colorado
//
//  Siena is free software: you can redistribute it and/or modify
//  it under the terms of the GNU General Public License as published by
//  the Free Software Foundation, either version 3 of the License, or
//  (at your option) any later version.
//  
//  Siena is distributed in the hope that it will be useful,
//  but WITHOUT ANY WARRANTY; without even the implied warranty of
//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
//  GNU General Public License for more details.
//  
//  You should have received a copy of the GNU General Public License
//  along with Siena.  If not, see <http://www.gnu.org/licenses/>.
//

//
//  this test must be withing package siena because Tokenizer is
//  private
//
package siena;

public class TestTokenizer {

    public static void main(String[] args) {
	try {
	    int res;
	    SENPBuffer t = new SENPBuffer();

	    while ((res = System.in.read(t.buf, 0, SENP.MaxPacketLen)) > 0) {
		t.init(res);
		int type;
		while((type = t.nextToken()) != SENPPacket.T_EOF) {
		    switch(type) {
		    case SENPPacket.T_INT:
			System.out.println("int: " + Long.toString(t.ival));
			break;
		    case SENPPacket.T_DOUBLE:
			System.out.println("double: "+Double.toString(t.dval));
			break;
		    case SENPPacket.T_STR:
			System.out.println("str: " + t.sval_string());
			break;
		    case SENPPacket.T_ID:
			System.out.println("id: " + t.sval_string());
			break;
		    case SENPPacket.T_BOOL:
			System.out.println("bool: " + t.bval);
			break;
		    case SENPPacket.T_OP:
			System.out.println("op: " + new String(SENP.operators[t.oval]));
			break;
		    case SENPPacket.T_LPAREN:
			System.out.println("{");
			break;
		    case SENPPacket.T_RPAREN:
			System.out.println("}");
			break;
		    case SENPPacket.T_UNKNOWN:
			System.out.println("unknown");
			break;
		    } 
		}
	    }
	} catch (Exception ex) {
	    ex.printStackTrace();
	}
    }
}
